Extension/src/LanguageServer/client.ts

```diff
@@ -961,6 +962,32 @@ export class DefaultClient implements Client {
     public onDidChangeSettings(event: vscode.ConfigurationChangeEvent): { [key: string] : string } {
         let changedSettings: { [key: string] : string } = this.settingsTracker.getChangedSettings();
         this.notifyWhenReady(() => {
+            // Unlike the LSP message, the event does not contain all settings as a payload, so we need to
+            // build a new JSON object with everything we need on the native side.
+            let settings: any = {
+                "C_Cpp": { ...vscode.workspace.getConfiguration("C_Cpp", this.RootUri) },
+                "files": vscode.workspace.getConfiguration("files", this.RootUri),
```

Comment:
Only the files.exclude, search.exclude, and editor.tabSize (missing) settings are needed.

Reply:
Where is editor.tabSize referenced?  In cpptools_didChangeSettings (was workspace_didChangeConfiguration) where the settings json is referenced, it's to get "C_Cpp", "files" and "search".